### PR TITLE
Add new transient registration type for NewRegistration

### PR DESCRIPTION
**Pull request by cintamani**
This is milestone 1 for our work on New registrations. This simply add a new model, along with an empty state machine, a basic test and an empty factory.

### CODE REVIEW

File: spec/models/waste_carriers_engine/new_registration_spec.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+module WasteCarriersEngine
+  RSpec.describe NewRegistration, type: :model do
+    subject(:renewing_registration) { build(:new_registration) }
```
Comment (cintamani) [inline spec/models/waste_carriers_engine/new_registration_spec.rb]:
```suggestion
    subject(:new_registration) { build(:new_registration) }
```
